coopr-docs/docs/source/guide/superadmin/chef-solo-automator-plugin.rst

```diff
@@ -141,25 +139,34 @@ service-level action data are merged together, and preserved at the top-level. C
     }
 
 
-Consider the following two rules of thumb:
-	* When using community cookbooks, attributes can be specified in Coopr templates exactly as the cookbook expects (at the top-level).
+Consider the following rules of thumb:
+	* When using community cookbooks, attributes can be specified in Coopr templates exactly as the cookbook expects (at the top-level). If separate services require the same recipe with different attributes, then these attributes can be specified in the service ``json_attributes`` field
 	* When writing cookbooks specifically utilizing Coopr metadata (cluster node data for example), recipes can access the metadata at ``node['coopr']['cluster']...``
 
 Bootstrap
 =========
 
 Each Coopr Automator plugin is responsible for implementing a bootstrap method in which it performs any actions it needs to be able to carry out further tasks. The Chef Solo Automator plugin performs the following actions for a bootstrap task:
-	1. Bundle its local copy of the cookbooks/roles/data_bags directories into tarballs, ``cookbooks.tar.gz``, ``roles.tar.gz``, ``data_bags.tar.gz``.
-		* Unless the tarballs exist already and were created in the last 10 minutes.
-	2. Logs into the remote box and installs chef via the Opscode Omnibus installer (``curl -L https://www.opscode.com/chef/install.sh | bash``).
-	3. Creates the remote coopr cache directory ``/var/cache/coopr``.
-	4. SCP the local tarballs to the remote Coopr cache directory.
-        5. Extracts the tarballs on the remote host to the default chef directory ``/var/chef``.
+	1. Bundle its local copy of the cookbooks/roles/data_bags directories into tarballs, ``cookbooks.tar.gz``, ``roles.tar.gz``, ``data_bags.tar.gz``
+		* Unless the tarballs exist already and were created in the last 10 minutes
+	2. Logs into the remote box and installs chef in one of three ways in this order:
+		a. via ``yum install`` to leverage any internal yum repositories preconfigured on the remote host
+		b. via ``apt-get install`` to leverage any internal apt repositories preconfigured on the remote host
+		c. via the Opscode Omnibus installer (``curl -L https://www.opscode.com/chef/install.sh | bash``). This requires internet access on the remote host
+	3. Creates the remote coopr cache directory ``/var/cache/coopr``
+	4. SCP the local tarballs to the remote Coopr cache directory
+        5. Extracts the tarballs on the remote host to the default chef directory ``/var/chef``
```

Comment:
remove a space before the "5" so that it lines up, otherwise it won't format correctly. ProTip: you can do the first item as  "1." and then the rest as "#.", and Sphinx will number them for you accordingly. As long as they line up. Handy if you add/subtract/move things...